### PR TITLE
Added setting to disable checking for auto updates.

### PR DESCRIPTION
**Pull request by Mikunj**
Users can manually disable checking for auto updates.
The reason we put `autoUpdate` in the user config and not the window storage is that anything in window storage will be wiped clear when user deletes the account (either manually or upon unlink). The auto update setting though shouldn't be per account but per device and thus we stick it in user config.

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -1090,6 +1090,22 @@ ipc.on('set-media-permissions', (event, value) => {
   }
 });
 
+// Loki - Auto updating
+ipc.on('get-auto-update-enabled', event => {
```
Comment (neuroscr) [inline main.js]: can we get the `-enabled` out of there? `get-auto-update_setting` would be an improvement (then saying get-auto-update-enabled is disabled)

File: main.js [hunk 2]
```diff
@@ -1090,6 +1090,22 @@ ipc.on('set-media-permissions', (event, value) => {
...
+  const configValue = userConfig.get('autoUpdate');
+  // eslint-disable-next-line no-param-reassign
+  event.returnValue = typeof configValue !== 'boolean' ? true : configValue;
+});
+
+ipc.on('set-auto-update-enabled', (event, value) => {
```
Comment (neuroscr) [inline main.js]: Same here `set-auto-update-setting` or just `set-auto-update`

File: main.js [hunk 3]
```diff
@@ -1090,6 +1090,22 @@ ipc.on('set-media-permissions', (event, value) => {
...
+  userConfig.set('autoUpdate', !!value);
+
+  // Stop updater if user turned it off
+  if (!value) {
+    updater.stop();
+  }
```
Comment (neuroscr) [inline main.js]: and if you turn it back on without restarting, should we start it up here too?

Comment (Mikunj) [reply main.js]: Good idea.

File: ts/updater/updater.ts [hunk 4]
```diff
@@ -25,28 +28,42 @@ export async function start(
...
   }, INTERVAL);
+  stopped = false;
 
   await checkForUpdates(getMainWindow, messages, logger);
 }
 
+export function stop() {
+  if (interval) {
+    clearInterval(interval);
+    stopped = true;
```
Comment (neuroscr) [inline ts/updater/updater.ts]: we should clear the interval variable here so it's not set, otherwise the if on line 31 will always trigger

Comment (Mikunj) [reply ts/updater/updater.ts]: Good catch

**Comment by Mikunj** [comment]
Ah i just realised i didn't fix it fully. I'll make the final fix and test tomorrow

**Pull request merged**